pkg/git/certificate_check_callback.go

```diff
@@ -1,40 +1,44 @@
 package git
 
 import (
-	git2go "github.com/libgit2/git2go/v31"
-
 	"log"
+
+	git2go "github.com/libgit2/git2go/v33"
+	"github.com/pkg/errors"
 )
 
 func certificateCheckCallback(logger *log.Logger) git2go.CertificateCheckCallback {
-	return func(cert *git2go.Certificate, valid bool, hostname string) git2go.ErrorCode {
+	return func(cert *git2go.Certificate, valid bool, hostname string) error {
 		if valid {
-			return git2go.ErrOk
+			return nil
 		}
 
 		if cert.Kind == git2go.CertificateX509 {
 			if cert.X509 != nil {
 				err := cert.X509.VerifyHostname(hostname)
 				if err != nil {
-					logger.Println("host name could not be verified")
-					return git2go.ErrAuth
+					msg := "host name could not be verified"
+					logger.Println(msg)
```

Comment:
Do we need a logger here now that the error message is being returned? As in will it still be printed?